src/Factory/FormAbstractServiceFactory.php

```diff
@@ -1,94 +1,101 @@
 <?php
-/**
- * @see https://github.com/dotkernel/dot-form/ for the canonical source repository
- * @copyright Copyright (c) 2017 Apidemia (https://www.apidemia.com)
- * @license https://github.com/dotkernel/dot-form/blob/master/LICENSE.md MIT License
- */
 
-declare(strict_types = 1);
+declare(strict_types=1);
 
 namespace Dot\Form\Factory;
 
-use Interop\Container\ContainerInterface;
-use Laminas\InputFilter\Factory;
+use Laminas\Filter\FilterPluginManager;
+use Laminas\Form\Factory;
+use Laminas\Form\Factory as FormFactory;
+use Laminas\Form\FormElementManager;
+use Laminas\Form\FormInterface;
 use Laminas\InputFilter\InputFilterInterface;
+use Laminas\InputFilter\InputFilterPluginManager;
 use Laminas\ServiceManager\Factory\AbstractFactoryInterface;
-use Laminas\Stdlib\ArrayUtils;
-use Laminas\Log\LoggerAbstractServiceFactory;
-
-/**
- * Class FormAbstractServiceFactory
- * @package Dot\Form\Factory
- */
-class FormAbstractServiceFactory extends LoggerAbstractServiceFactory
+use Laminas\Validator\ValidatorPluginManager;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+use function count;
+use function explode;
+use function is_array;
+use function is_string;
+
+class FormAbstractServiceFactory implements AbstractFactoryInterface
 {
-    const PREFIX = 'dot-form';
+    public const PREFIX = 'dot-form';
+
+    private ?array $config = null;
+
+    protected string $configKey = 'dot_form';
 
-    /** @var string */
-    protected $configKey = 'dot_form';
+    protected string $subConfigKey = 'forms';
 
-    /** @var string */
-    protected $subConfigKey = 'forms';
+    private ?FormFactory $factory = null;
 
     /**
-     * @param ContainerInterface $container
      * @param string $requestedName
-     * @return bool
      */
-    public function canCreate(ContainerInterface $container, $requestedName)
+    public function canCreate(ContainerInterface $container, $requestedName): bool
     {
         $parts = explode('.', $requestedName);
         if (count($parts) !== 2) {
             return false;
         }
+
         if ($parts[0] !== static::PREFIX) {
             return false;
         }
-        return parent::canCreate($container, $parts[1]);
+
+        // avoid infinite loops when looking up config
+        if ($requestedName === 'config') {
+            return false;
+        }
+
+        $config = $this->getConfig($container);
+        if (empty($config)) {
+            return false;
+        }
+
+        return isset($config[$requestedName]) && is_array($config[$requestedName]) && ! empty($config[$requestedName]);
```

Comment:
```suggestion
        return ! empty($config[$requestedName]) && is_array($config[$requestedName]);
```
`isset` is unnecessary because it is checked by `! empty`